### PR TITLE
Improve pixel classifier

### PR DESCRIPTION
**Pull request by jgersak**
syntax shortening

### CODE REVIEW

File: s2cloudless/pixel_classifier.py [hunk 1]
```diff
@@ -63,9 +64,7 @@ def extract_pixels(data: np.ndarray) -> np.ndarray:
                 "[n_images, n_pixels_y, n_pixels_x, n_bands]"
             )
 
-        new_shape = data.shape[0] * data.shape[1] * data.shape[2], data.shape[3]
-        pixels = data.reshape(new_shape)
-        return pixels
+        return data.reshape(data.shape[0] * data.shape[1] * data.shape[2], data.shape[3])
```
Comment (zigaLuksic) [inline s2cloudless/pixel_classifier.py:67]: try `data.reshape((-1, data.shape[3]))`

File: s2cloudless/pixel_classifier.py [hunk 2]
```diff
@@ -103,4 +102,4 @@ def image_predict_proba(self, data: np.ndarray, **kwargs: Any) -> np.ndarray:
         else:
             probabilities = self.classifier.predict_proba(pixels, **kwargs)
 
-        return probabilities.reshape(data.shape[0], data.shape[1], data.shape[2], probabilities.shape[1])
+        return probabilities.reshape(*data.shape[0:3], probabilities.shape[1])
```
Comment (zigaLuksic) [inline s2cloudless/pixel_classifier.py:105]: i'd go with `probabilities.shape[-1]`

File: s2cloudless/pixel_classifier.py [hunk 3]
```diff
@@ -30,16 +30,15 @@ def __init__(self, classifier: Any):
         """
         :param classifier: An instance of trained classifier that will be executed over an entire image
         """
-        self._check_classifier(classifier)
-        self.classifier = classifier
+        self.classifier = self._check_classifier(classifier)
```
Comment (zigaLuksic) [inline s2cloudless/pixel_classifier.py:33]: not a fan of having `_check_classifier` return a classifier, it doesn't suit the name.
as for the rest, this way you actually have one extra line in the code :)

Comment (jgersak) [reply s2cloudless/pixel_classifier.py]: reverted and updated, can not remove Booster check in _check_classifier

File: s2cloudless/pixel_classifier.py (hunk 1 above)
Comment (jgersak) [reply s2cloudless/pixel_classifier.py]: works, done

File: s2cloudless/pixel_classifier.py [hunk 4]
```diff
@@ -42,12 +42,9 @@ def _check_classifier(classifier: Any) -> None:
...
 
         predict = getattr(classifier, "predict", None)
-        if not callable(predict):
-            raise ValueError("Classifier does not have a predict method!")
-
         predict_proba = getattr(classifier, "predict_proba", None)
-        if not callable(predict_proba):
-            raise ValueError("Classifier does not have a predict_proba method!")
+        if not callable(predict) or not callable(predict_proba):
+            raise ValueError("Classifier does not have a predict or predict_proba method!")
```
Comment (zigaLuksic) [inline s2cloudless/pixel_classifier.py:47]: try
```python
for method_name in ("predict", "predict_proba"):
    method = getattr(classifier, method_name, None)
    if not callable(method):
        raise ValueError(f"Classifier does not have a {method_name} method!")
```

Comment (jgersak) [reply s2cloudless/pixel_classifier.py]: Works fine, looks great, used.

**Pull request merged**